### PR TITLE
feat(curriculum): create basic css quiz

### LINKED ISSUE #56179: [HACKTOBERFEST 2024] - Basic CSS Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Basic CSS quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-basic-css`.

**All quiz questions need to relate to one of these topics:**

- what is css
- basic anatomy/structure of a css rule
- meta viewport
- inline vs internal vs external css
- width and height
- descendant combinator
- child combinator (>)
- next-sibling combinator (+) 
- subsequent-sibling combinator (~)
- inline vs block elements
- What is inline-block
- css specificity (class, inline, internal, external styles, ids, type selectors)
- What is the * selector
- !important
- how does css cascade algorithm work
- margin 
- padding
- line-height

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by clydehenry3**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56179

### CODE REVIEW

**Comment by clydehenry3** [comment]
Hey @naomi-lgbt! My 7 day deadline is tomorrow from issue #56179. If the PR isn't accepted by then, will this kick me out of that issue and lose my credit?

Thanks!

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 1]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Cascading Style Sheets
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which answer uses the correct strucutre of using a CSS rule?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:42]:
```suggestion
Which answer uses the correct structure of using a CSS rule?
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 2]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`p {color: red;}`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does the `<meta>` viewport do?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:64]:
```suggestion
What does `<meta name="viewport">` do?
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 3]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`<p style="color: blue;"></p>`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+When using internal CSS, where is the `<style>` placed within the HTML?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:108]:
```suggestion
When using internal CSS, where is the `style` element placed within the HTML?
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 4]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --distractors--
 
-Placeholder distractor 1
+`<meta>`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:112]:
```suggestion
In the `meta` element
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 5]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+`<script>`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:116]:
```suggestion
In the `script` element
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 6]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+`<body>`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:120]:
```suggestion
In the `body` element
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 7]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+`<head>`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:124]:
```suggestion
In the `head` element
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 8]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`width: 50px; height: 50px;`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which rule is correct to target `<h1>` inside of a `<div>`?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:152]:
```suggestion
Which selector is correct to target `h1` elements inside of a `div`?
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 9]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`div h1 {}`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which rule is correct to target direct children of a `<footer>`?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:174]:
```suggestion
Which selector is correct to target direct children of a `footer`?
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 10]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`footer > ul {}`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which rule is correct to target the next sibling of an `<img>`?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:196]:
```suggestion
Which selector is correct to target the next sibling of an `img`?
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 11]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`img + h1 {}`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which rule is correct to target all siblings of an `<img>`?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:218]:
```suggestion
Which selector is correct to target all siblings of an `img`?
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 12]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Blockline elements start on a new line with margin before and after the element.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What statement is TRUE when using the `inline-block` rule?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:262]:
```suggestion
What statement is TRUE when using the `inline-block` value?
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 13]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+Allows you to place elements next to each other while letting you set the height and width.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Given the following selectors, which has the highest priority?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:284]:
```suggestion
Given the following selectors, which has the highest specificity?
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 14]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`#id`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Given the following selectors, which has the lowest priority?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:306]:
```suggestion
Given the following selectors, which has the lowest specificity?
```

The answers also don't match. Type selectors have the same specificity - `div` is no more the correct answer than `h1` is.

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 15]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+Targets elements that have children on the page.
 
 ---
 
-Placeholder distractor 3
+Targets all `<p>` elements on the page.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:340]:
```suggestion
Targets all `p` elements on the page.
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 16]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+Applies on to a certain selector or group of elements.
 
 #### --answer--
 
-Placeholder answer
+Overrides any other styles that the property is used on.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:366]:
```suggestion
Overrides any other values applied to the property for that selector.
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 17]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+Improves Search Emgine Optimiztion (SEO).
 
 #### --answer--
 
-Placeholder answer
+Determines styles of the element based on specificity and order of they appear from top to bottom.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:388]:
```suggestion
Determines styles of the element based on specificity and order of declaration.
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 18]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which rule applies 32px of margin to all sides?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:394]:
```suggestion
Which rule applies `32px` of margin to all sides?
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 19]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+`margin: 32px;`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which rule applies 24px padding to the top and bottom?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:416]:
```suggestion
Which rule applies `24px` padding to the top and bottom?
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 20]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What does the `line-height` property do?
 
 #### --distractors--
 
-Placeholder distractor 1
+Controls how size of the font.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:442]:
```suggestion
Controls the size of the font.
```

File: curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md [hunk 21]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+Controls how much space is between the letters of text.
 
 #### --answer--
 
-Placeholder answer
-
+Controls how much space is between the lines of text.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-basic-css/66ed8fa2f45ce3ece4053eab.md:454]:
```suggestion
Controls how much space is between the lines of text.

```

Restore the EOF new line.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Review by clydehenry3 — COMMENTED** [review]
Thank you so much for reviewing my code @naomi-lgbt!

**Comment by naomi-lgbt** [comment]
image

Looks like the ones that are "hidden" got missed.

**Comment by clydehenry3** [comment]
> image
> 
> Looks like the ones that are "hidden" got missed.

My bad! Those were now resolved.

**Review by naomi-lgbt — APPROVED** [review]

**Review by gikf — APPROVED** [review]

**Review by gikf — APPROVED** [review]

**Comment by gikf** [comment]
Hi @clydehenry3. Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**